### PR TITLE
PT-159960482 Test that minimum oracle response tx fee depends on orac…

### PR DESCRIPTION
**Pull request by zp-sd**
…le response size

### CODE REVIEW

File: apps/aeoracle/test/aeoracle_SUITE.erl [hunk 1]
```diff
@@ -462,6 +464,38 @@ query_response(Cfg, QueryOpts) ->
...
+    size(SmallResponse) < size(BiggerResponse),
+
+    %% Deduce minimal fee for response tx with SmallResponse
+    RTx0        = aeo_test_utils:response_tx(OracleKey, ID, SmallResponse, #{}, S1),
+    MinimalFee  = aetx:min_fee(RTx0),
+
+    %% Test oracle response tx with SmallResponse is accepted with MinimalFee
+    RTx1        = aeo_test_utils:response_tx(OracleKey, ID, SmallResponse, #{fee => MinimalFee}, S1),
+    MinimalFee1 = MinimalFee,
+    {ok, Trees} = aetx:check(RTx1, Trees, Env),
```
Comment (lucafavatella) [inline apps/aeoracle/test/aeoracle_SUITE.erl]: This assertion is tricky, as a greater fee value itself may enlarge the tx serialization hence increase the minimum fee. Though test seems stable enough, and can be fine tuned if/when needed if fee crosses some threshold. Ok.

File: apps/aeoracle/test/aeoracle_SUITE.erl [hunk 2]
```diff
@@ -462,6 +464,38 @@ query_response(Cfg, QueryOpts) ->
...
+
+    %% Test oracle response tx with BiggerResponse is not accepted with MinimalFee
+    RTx2        = aeo_test_utils:response_tx(OracleKey, ID, BiggerResponse, #{fee => MinimalFee}, S1),
+    MinimalFee2 = aetx:min_fee(RTx2),
+    MinimalFee2 > MinimalFee1,
```
Comment (lucafavatella) [inline apps/aeoracle/test/aeoracle_SUITE.erl]: Is this meant to be assertion?

File: apps/aeoracle/test/aeoracle_SUITE.erl [hunk 3]
```diff
@@ -462,6 +464,37 @@ query_response(Cfg, QueryOpts) ->
...
     {OracleKey, ID, S2}.
 
+query_response_fee_depends_on_response_size(Cfg) ->
+    {OracleKey, ID, S1} = query_oracle(Cfg, #{}, #{}),
+    Trees               = aeo_test_utils:trees(S1),
+    Env                 = aetx_env:tx_env(?ORACLE_RSP_HEIGHT),
+
+    SmallResponse  = <<"small_response">>,
+    BiggerResponse = <<"bigger_oracle_response">>,
+    size(SmallResponse) < size(BiggerResponse),
```
Comment (lucafavatella) [inline apps/aeoracle/test/aeoracle_SUITE.erl]: Is this meant to be assertion?

File: apps/aeoracle/test/aeoracle_SUITE.erl [hunk 4]
```diff
@@ -462,6 +464,37 @@ query_response(Cfg, QueryOpts) ->
...
+    %% Test oracle response tx with BiggerResponse is not accepted with MinimalFee
+    RTx2        = aeo_test_utils:response_tx(OracleKey, ID, BiggerResponse, #{fee => MinimalFee}, S1),
+    MinimalFee2 = aetx:min_fee(RTx2),
+    MinimalFee2 > MinimalFee1,
+    {error, too_low_fee} = aetx:check(RTx2, Trees, Env),
+
+    %% Test oracle response tx with BiggerResponse is accepted with MinimalFee2
+    RTx3        = aeo_test_utils:response_tx(OracleKey, ID, BiggerResponse, #{fee => MinimalFee2}, S1),
+    MinimalFee3 = aetx:min_fee(RTx3),
+    MinimalFee3 = MinimalFee2,
```
Comment (lucafavatella) [inline apps/aeoracle/test/aeoracle_SUITE.erl]: I see `MinimalFee3` bound so works as assertion. Ok.

File: apps/aeoracle/test/aeoracle_SUITE.erl [hunk 5]
```diff
@@ -462,6 +464,37 @@ query_response(Cfg, QueryOpts) ->
...
+    {ok, Trees} = aetx:check(RTx3, Trees, Env),
```
Comment (lucafavatella) [inline apps/aeoracle/test/aeoracle_SUITE.erl]: Again - tricky but stable enough so ok.

File: apps/aeoracle/test/aeoracle_SUITE.erl (hunk 2 above)
Comment (zp-sd) [reply apps/aeoracle/test/aeoracle_SUITE.erl]: Yes, ☕️ made me realize it is not going to work :)

File: apps/aeoracle/test/aeoracle_SUITE.erl (hunk 3 above)
Comment (zp-sd) [reply apps/aeoracle/test/aeoracle_SUITE.erl]: Yes, ☕️ made me realize it is not going to work :)

File: apps/aeoracle/test/aeoracle_SUITE.erl (hunk 1 above)
Comment (zp-sd) [reply apps/aeoracle/test/aeoracle_SUITE.erl]: Yes, the other option is to just get minimal fee of transactions with `SmallResponse` and `BiggerResponse`, and compare them. But this should be easy enough to adjust if we encounter problems with the test.

**Comment by lucafavatella** [comment]
Please prefer merging (as opposed to squash or rebase) and not force pushing (if possible) in order not to require rebase of #1792

**Pull request merged**